CI failure: `package slices is not in GOROOT`

```

=== Failed
=== FAIL: cmd/nerdctl/container TestRunWithLogBinary (18.58s)
    testutil.go:598: buildkitHost="unix:///run/buildkit-nerdctl-test/buildkitd.sock"
    container_run_test.go:540: assertion failed: res.ExitCode is not exitCode: 
        Command:  /usr/local/bin/nerdctl --namespace=nerdctl-test build /tmp/TestRunWithLogBinary1321443321/001 --output type=local,src=/go/src/logger/logger,dest=/tmp/TestRunWithLogBinary1321443321/002
        ExitCode: 1
        Error:    exit status 1
        Stdout:   
        Stderr:   #1 [internal] load build definition from Dockerfile
        #1 transferring dockerfile: 1.32kB done
        #1 WARN: NoEmptyContinuation: Empty continuation line (line 48)
        #1 WARN: FromAsCasing: 'as' and 'FROM' keywords' casing do not match (line 2)
        #1 DONE 0.0s
        
        #2 [internal] load metadata for ghcr.io/stargz-containers/golang:1.18-org
        #2 DONE 0.8s
        
        #3 [internal] load .dockerignore
        #3 transferring context: 2B done
        #3 DONE 0.0s
        
        #4 [builder 1/8] FROM ghcr.io/stargz-containers/golang:1.18-org@sha256:b5bb76229c16712311992bd6c794eb6b96c95320a20cab2e24cce862c2659362
        #4 resolve ghcr.io/stargz-containers/golang:1.18-org@sha256:b5bb76229c16712311992bd6c794eb6b96c95320a20cab2e24cce862c2659362 0.0s done
        #4 sha256:e795b75807e4733c35ed0199e16944199899da6f4c9e9824552e711f85f204c0 0B / 156B 0.2s
        #4 sha256:e795b75807e4733c35ed0199e16944199899da6f4c9e9824552e711f85f204c0 156B / 156B 0.2s done
        #4 sha256:efc242f34a7ed1a729a1063a18fba314d6e3667da40b1f36fdfeda9bdbb2824e 5.24MB / 141.70MB 0.3s
        #4 sha256:26ee1eb1b3b371b2e378312e42f25ae777a3276e52b3d105b2512d1ef852431d 0B / 85.83MB 0.2s
        #4 sha256:d751dc38ae511bbc21c148bffa7e863057cbc7b4a8ff5155f2eca7e8d03740c6 0B / 54.58MB 0.2s
        #4 sha256:efc242f34a7ed1a729a1063a18fba314d6e3667da40b1f36fdfeda9bdbb2824e 13.63MB / 141.70MB 0.5s
        #4 sha256:d751dc38ae511bbc21c148bffa7e863057cbc7b4a8ff5155f2eca7e8d03740c6 11.31MB / 54.58MB 0.3s
        #4 sha256:a0bf850a0df065fb202ebf8a3527699dc18322469c34733a6cb7f412a7aaefa6 0B / 10.87MB 0.2s
        #4 sha256:d751dc38ae511bbc21c148bffa7e863057cbc7b4a8ff5155f2eca7e8d03740c6 20.43MB / 54.58MB 0.5s
        #4 sha256:a0bf850a0df065fb202ebf8a3527699dc18322469c34733a6cb7f412a7aaefa6 10.87MB / 10.87MB 0.4s done
        #4 sha256:efc242f34a7ed1a729a1063a18fba314d6e3667da40b1f36fdfeda9bdbb2824e 26.21MB / 141.70MB 0.6s
        #4 sha256:26ee1eb1b3b371b2e378312e42f25ae777a3276e52b3d105b2512d1ef852431d 12.58MB / 85.83MB 0.6s
        #4 sha256:d751dc38ae511bbc21c148bffa7e863057cbc7b4a8ff5155f2eca7e8d03740c6 29.36MB / 54.58MB 0.6s
        #4 sha256:540ff8c0841d610e4cc2ad3b9ed4c6edcad4f5be2add8765f416515fbc2be2a8 0B / 5.15MB 0.2s
        #4 sha256:efc242f34a7ed1a729a1063a18fba314d6e3667da40b1f36fdfeda9bdbb2824e 33.55MB / 141.70MB 0.8s
        #4 sha256:26ee1eb1b3b371b2e378312e42f25ae777a3276e52b3d105b2512d1ef852431d 20.97MB / 85.83MB 0.8s
        #4 sha256:d751dc38ae511bbc21c148bffa7e863057cbc7b4a8ff5155f2eca7e8d03740c6 37.75MB / 54.58MB 0.8s
        #4 sha256:540ff8c0841d610e4cc2ad3b9ed4c6edcad4f5be2add8765f416515fbc2be2a8 3.15MB / 5.15MB 0.3s
        #4 sha256:efc242f34a7ed1a729a1063a18fba314d6e3667da40b1f36fdfeda9bdbb2824e 43.41MB / 141.70MB 0.9s
        #4 sha256:26ee1eb1b3b371b2e378312e42f25ae777a3276e52b3d105b2512d1ef852431d 29.36MB / 85.83MB 0.9s
        #4 sha256:d751dc38ae511bbc21c148bffa7e863057cbc7b4a8ff5155f2eca7e8d03740c6 48.08MB / 54.58MB 0.9s
        #4 sha256:540ff8c0841d610e4cc2ad3b9ed4c6edcad4f5be2add8765f416515fbc2be2a8 5.15MB / 5.15MB 0.4s done
        #4 sha256:efc242f34a7ed1a729a1063a18fba314d6e3667da40b1f36fdfeda9bdbb2824e 57.67MB / 141.70MB 1.1s
        #4 sha256:26ee1eb1b3b371b2e378312e42f25ae777a3276e52b3d105b2512d1ef852431d 39.85MB / 85.83MB 1.1s
        #4 sha256:d751dc38ae511bbc21c148bffa7e863057cbc7b4a8ff5155f2eca7e8d03740c6 54.58MB / 54.58MB 1.0s done
        #4 sha256:5492f66d270062ddb73f28649d80eef162f2c9376d53973a3557158390af4f30 761.86kB / 54.92MB 0.2s
        #4 sha256:efc242f34a7ed1a729a1063a18fba314d6e3667da40b1f36fdfeda9bdbb2824e 69.21MB / 141.70MB 1.2s
        #4 sha256:26ee1eb1b3b371b2e378312e42f25ae777a3276e52b3d105b2512d1ef852431d 51.38MB / 85.83MB 1.2s
        #4 sha256:5492f66d270062ddb73f28649d80eef162f2c9376d53973a3557158390af4f30 11.53MB / 54.92MB 0.3s
        #4 sha256:efc242f34a7ed1a729a1063a18fba314d6e3667da40b1f36fdfeda9bdbb2824e 79.69MB / 141.70MB 1.4s
        #4 sha256:26ee1eb1b3b371b2e378312e42f25ae777a3276e52b3d105b2512d1ef852431d 61.87MB / 85.83MB 1.4s
        #4 sha256:5492f66d270062ddb73f28649d80eef162f2c9376d53973a3557158390af4f30 22.02MB / 54.92MB 0.5s
        #4 sha256:efc242f34a7ed1a729a1063a18fba314d6e3667da40b1f36fdfeda9bdbb2824e 90.18MB / 141.70MB 1.5s
        #4 sha256:26ee1eb1b3b371b2e378312e42f25ae777a3276e52b3d105b2512d1ef852431d 74.45MB / 85.83MB 1.5s
        #4 sha256:5492f66d270062ddb73f28649d80eef162f2c9376d53973a3557158390af4f30 32.51MB / 54.92MB 0.6s
        #4 sha256:efc242f34a7ed1a729a1063a18fba314d6e3667da40b1f36fdfeda9bdbb2824e 101.71MB / 141.70MB 1.7s
        #4 sha256:26ee1eb1b3b371b2e378312e42f25ae777a3276e52b3d105b2512d1ef852431d 85.83MB / 85.83MB 1.7s done
        #4 sha256:efc242f34a7ed1a729a1063a18fba314d6e3667da40b1f36fdfeda9bdbb2824e 112.20MB / 141.70MB 1.8s
        #4 sha256:5492f66d270062ddb73f28649d80eef162f2c9376d53973a3557158390af4f30 54.92MB / 54.92MB 0.9s
        #4 sha256:efc242f34a7ed1a729a1063a18fba314d6e3667da40b1f36fdfeda9bdbb2824e 134.22MB / 141.70MB 2.1s
        #4 sha256:5492f66d270062ddb73f28649d80eef162f2c9376d53973a3557158390af4f30 54.92MB / 54.92MB 1.0s done
        #4 extracting sha256:5492f66d270062ddb73f28649d80eef162f2c9376d53973a3557158390af4f30
        #4 sha256:efc242f34a7ed1a729a1063a18fba314d6e3667da40b1f36fdfeda9bdbb2824e 141.70MB / 141.70MB 2.3s
        #4 sha256:efc242f34a7ed1a729a1063a18fba314d6e3667da40b1f36fdfeda9bdbb2824e 141.70MB / 141.70MB 2.3s done
        #4 extracting sha256:5492f66d270062ddb73f28649d80eef162f2c9376d53973a3557158390af4f30 2.1s done
        #4 DONE 4.1s
        
        #4 [builder 1/8] FROM ghcr.io/stargz-containers/golang:1.18-org@sha256:b5bb76229c16712311992bd6c794eb6b96c95320a20cab2e24cce862c2659362
        #4 extracting sha256:540ff8c0841d610e4cc2ad3b9ed4c6edcad4f5be2add8765f416515fbc2be2a8
        #4 extracting sha256:540ff8c0841d610e4cc2ad3b9ed4c6edcad4f5be2add8765f416515fbc2be2a8 0.3s done
        #4 extracting sha256:a0bf850a0df065fb202ebf8a3527699dc18322469c34733a6cb7f412a7aaefa6
        #4 extracting sha256:a0bf850a0df065fb202ebf8a3527699dc18322469c34733a6cb7f412a7aaefa6 0.2s done
        #4 DONE 4.6s
        
        #4 [builder 1/8] FROM ghcr.io/stargz-containers/golang:1.18-org@sha256:b5bb76229c16712311992bd6c794eb6b96c95320a20cab2e24cce862c2659362
        #4 extracting sha256:d751dc38ae511bbc21c148bffa7e863057cbc7b4a8ff5155f2eca7e8d03740c6
        #4 extracting sha256:d751dc38ae511bbc21c148bffa7e863057cbc7b4a8ff5155f2eca7e8d03740c6 1.9s done
        #4 DONE 6.4s
        
        #4 [builder 1/8] FROM ghcr.io/stargz-containers/golang:1.18-org@sha256:b5bb76229c16712311992bd6c794eb6b96c95320a20cab2e24cce862c2659362
        #4 extracting sha256:26ee1eb1b3b371b2e378312e42f25ae777a3276e52b3d105b2512d1ef852431d
        #4 extracting sha256:26ee1eb1b3b371b2e378312e42f25ae777a3276e52b3d105b2512d1ef852431d 2.0s done
        #4 DONE 8.4s
        
        #4 [builder 1/8] FROM ghcr.io/stargz-containers/golang:1.18-org@sha256:b5bb76229c16712311992bd6c794eb6b96c95320a20cab2e24cce862c2659362
        #4 extracting sha256:efc242f34a7ed1a729a1063a18fba314d6e3667da40b1f36fdfeda9bdbb2824e
        #4 extracting sha256:efc242f34a7ed1a729a1063a18fba314d6e3667da40b1f36fdfeda9bdbb2824e 6.3s done
        #4 DONE 14.7s
        
        #4 [builder 1/8] FROM ghcr.io/stargz-containers/golang:1.18-org@sha256:b5bb76229c16712311992bd6c794eb6b96c95320a20cab2e24cce862c2659362
        #4 extracting sha256:e795b75807e4733c35ed0199e16944199899da6f4c9e9824552e711f85f204c0 done
        #4 DONE 14.7s
        
        #5 [builder 2/8] WORKDIR /go/src/
        #5 DONE 0.3s
        
        #6 [builder 3/8] RUN mkdir -p logger
        #6 DONE 0.1s
        
        #7 [builder 4/8] WORKDIR /go/src/logger
        #7 DONE 0.0s
        
        #8 [builder 5/8] RUN echo '	package main \n	\n	import ( \n	"bufio" \n	"context" \n	"fmt" \n	"io" \n	"os" \n	"path/filepath" \n	"sync" \n	\n	"github.com/containerd/containerd/v2/core/runtime/v2/logging"\n	)\n	func main() {\n		logging.Run(log)\n	}\n	func log(ctx context.Context, config *logging.Config, ready func() error) error {\n		var wg sync.WaitGroup \n		wg.Add(2) \n		// forward both stdout and stderr to temp files \n		go copy(&wg, config.Stdout, config.ID, "stdout") \n		go copy(&wg, config.Stderr, config.ID, "stderr") \n		// signal that we are ready and setup for the container to be started \n		if err := ready(); err != nil { \n		return err \n		} \n		wg.Wait() \n		return nil \n	}\n	\n	func copy(wg *sync.WaitGroup, r io.Reader, id string, kind string) { \n		f, _ := os.Create(filepath.Join(os.TempDir(), fmt.Sprintf("%s_%s.log", id, kind))) \n		defer f.Close() \n		defer wg.Done() \n		s := bufio.NewScanner(r) \n		for s.Scan() { \n			f.WriteString(s.Text()) \n		} \n	}\n' >> main.go
        #8 DONE 0.1s
        
        #9 [builder 6/8] RUN go mod init
        #9 0.078 go: creating new go.mod: module logger
        #9 0.081 go: to add module requirements and sums:
        #9 0.081 	go mod tidy
        #9 DONE 0.1s
        
        #10 [builder 7/8] RUN go mod tidy
        #10 0.122 go: finding module for package github.com/containerd/containerd/v2/core/runtime/v2/logging
        #10 0.584 go: downloading github.com/containerd/containerd/v2 v2.1.0
        #10 1.064 go: downloading github.com/containerd/containerd v1.7.27
        #10 1.516 go: found github.com/containerd/containerd/v2/core/runtime/v2/logging in github.com/containerd/containerd/v2 v2.1.0
        #10 1.519 go: downloading github.com/Microsoft/go-winio v0.6.2
        #10 1.519 go: downloading golang.org/x/sys v0.33.0
        #10 DONE 2.0s
        
        #11 [builder 8/8] RUN go build .
        #11 0.268 /go/pkg/mod/golang.org/x/sys@v0.33.0/unix/syscall_linux.go:16:2: package slices is not in GOROOT (/usr/local/go/src/slices)
        #11 ERROR: process "/bin/sh -c go build ." did not complete successfully: exit code: 1
        ------
         > [builder 8/8] RUN go build .:
        0.268 /go/pkg/mod/golang.org/x/sys@v0.33.0/unix/syscall_linux.go:16:2: package slices is not in GOROOT (/usr/local/go/src/slices)
        ------
        Dockerfile:53
        --------------------
          51 |     RUN go mod init
          52 |     RUN go mod tidy
          53 | >>> RUN go build .
          54 |     
          55 |     FROM scratch
        --------------------
        error: failed to solve: process "/bin/sh -c go build ." did not complete successfully: exit code: 1
        
```

https://github.com/containerd/nerdctl/actions/runs/14892500390/job/41828122104

--- a/cmd/nerdctl/container/container_run_test.go
+++ b/cmd/nerdctl/container/container_run_test.go
@@ -528,6 +528,8 @@ RUN echo '\
 
 
 RUN go mod init
+# Workaround for "package slices is not in GOROOT" https://github.com/containerd/nerdctl/issues/4214
+RUN go get github.com/containerd/containerd/v2@v2.0.5
 RUN go mod tidy
 RUN go build .
 

